ksql-metastore/src/main/java/io/confluent/ksql/metastore/MetaStoreImpl.java

```diff
@@ -250,6 +239,18 @@ public AggregateFunctionFactory getAggregateFactory(final String functionName) {
     return functionRegistry.listAggregateFunctions();
   }
 
+  private Stream<SourceInfo> streamSources(final Set<String> sourceNames) {
+    return sourceNames.stream()
+        .map(sourceName -> {
+          final SourceInfo sourceInfo = dataSources.get(sourceName);
+          if (sourceInfo == null) {
+            throw new KsqlException("Unknown source: " + sourceName);
```

Comment:
I'm not worried about the overhead of checking each source exists _twice_. (Not four times, as there are two different sets of streams and each is checked twice). 

Nor am I worried about lazy stream evaluation or throwing exceptions from streams. 

Yes, we could throw a message with all the missing streams, but I think the ROI is limited. The check never even existed before and we've not had any issues reported. Calls would previously have failed with an NPE.  This exception is thrown if a source exists when a statement is first being executed, but is dropped before the new query is registered.  

In short, I prefer this more succinct code that explicit checking up front.

Reply:
Yeah, I'm not worried about the performance either - it was just an added benefit. I still prefer the "up front" approach (I think it may actually end up being fewer lines of code) but it's purely subjective so we can defer to your approach since you're doing the hard coding work 😝